test/OpenSpaceToolkit/Astrodynamics/Solver/LeastSquaresSolver.test.cpp

```diff
@@ -0,0 +1,422 @@
+        EXPECT_THROW(LeastSquaresSolver(20, -1.0), ostk::core::error::RuntimeError);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver, Getters)
+{
+    {
+        EXPECT_EQ(solver_.getMaxIterationCount(), maxIterationCount_);
+        EXPECT_EQ(solver_.getRmsUpdateThreshold(), rmsUpdateThreshold_);
+        EXPECT_NO_THROW(solver_.getFiniteDifferenceSolver());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver, Solve_Success)
+{
+    // Test case where a priori is ignored and all observations weighted the same
+    {
+        const LeastSquaresSolver::Analysis analysis =
+            solver_.solve(initialGuessState_, observations_, generateStates_, {}, {});
+
+        // Check results
+        EXPECT_EQ(analysis.terminationCriteria, "RMS Update Threshold");
+        EXPECT_LT(analysis.rmsError, 0.1);
+        EXPECT_LT(analysis.iterationCount, solver_.getMaxIterationCount());
+
+        const VectorXd estimatedPosition = analysis.solutionState.getPosition().getCoordinates();
+        const VectorXd estimatedVelocity = analysis.solutionState.getVelocity().getCoordinates();
+
+        const VectorXd truePosition = trueState_.getPosition().getCoordinates();
+        const VectorXd trueVelocity = trueState_.getVelocity().getCoordinates();
+
+        EXPECT_LT((estimatedPosition - truePosition).norm(), 0.15);
+        EXPECT_LT((estimatedVelocity - trueVelocity).norm(), 0.15);
+    }
+
+    // Test case with weighted observations
+    {
+        const LeastSquaresSolver::Analysis analysis =
+            solver_.solve(initialGuessState_, observations_, generateStates_, initialStateSigmas_, observationSigmas_);
+
+        EXPECT_EQ(analysis.terminationCriteria, "RMS Update Threshold");
+        EXPECT_LT(analysis.rmsError, 20.0);
+        EXPECT_LT(analysis.iterationCount, solver_.getMaxIterationCount());
+
+        const VectorXd estimatedPosition = analysis.solutionState.getPosition().getCoordinates();
+        const VectorXd estimatedVelocity = analysis.solutionState.getVelocity().getCoordinates();
+        const VectorXd truePosition = trueState_.getPosition().getCoordinates();
+        const VectorXd trueVelocity = trueState_.getVelocity().getCoordinates();
+
+        EXPECT_LT((estimatedPosition - truePosition).norm(), 0.15);
+        EXPECT_LT((estimatedVelocity - trueVelocity).norm(), 0.15);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver, Solve_Failures)
+{
+    // Test undefined initial guess state
+    {
+        EXPECT_THROW(
+            solver_.solve(State::Undefined(), observations_, generateStates_), ostk::core::error::runtime::Undefined
+        );
+    }
+
+    // Test empty observations
+    {
+        EXPECT_THROW(
+            solver_.solve(initialGuessState_, Array<State>::Empty(), generateStates_),
+            ostk::core::error::runtime::Undefined
+        );
+    }
```

Comment:
If you agree with deleting the if statement that corresponds to this test, you can add one that corresponds the matrix dimension condition (to yield a full rank H matri)